### PR TITLE
[SE-0289] Revise propose per Core Team discussion

### PR DESCRIPTION
**Pull request by DougGregor**

### CODE REVIEW

File: proposals/0289-result-builders.md [hunk 1]
```diff
@@ -1124,3 +1185,64 @@ extension HTMLDocument {
...
+However, it will reject any statements that don't produce a `String`. What if the builder wanted to allow other statements, but without collecting the values? One could re-work the builder like this:
+
+```swift
+@resultBuilder
+struct StringConcatenator {
+  static func buildExpression(_ string: String) -> String? { string }
+  static func buildExpression<T>(_: T) -> String? { nil }
+
+  static func buildBlock(_ strings: String?...) -> String { 
+    strings.flatMap({$0}).joined(separator: ", ")
```
Comment (benrimmington) [inline proposals/0289-result-builders.md:1213]:
```suggestion
    strings.compactMap({$0}).joined(separator: ", ")
```

**Review by benrimmington — COMMENTED** [review]
Other suggestions:

* "function-building method(s)" → "result-building method(s)" (9 occurrences)

* `FunctionBuilderTerm` → `ResultBuilderTerm` (6 occurrences)

* `FunctionBuilder` → `ResultBuilder` (5 occurrences)

File: proposals/0289-result-builders.md [hunk 2]
```diff
@@ -240,47 +250,98 @@ These last two points (and some other considerations) strongly suggest that the
...
+  /// which defaults to Component if buildExpression() is not provided.
+  typealias Expression = ...
+
+  /// The type of a partial result, which will be carried through all of the
+  /// build functions.
+  typealias Component = ...
+
+  /// The type of the final returned result, which defaults to Component if
+  /// buildFinalResult() is not provided.
+  typealias Result = ...
```
[the remark this answers is not in the document]
Comment (DougGregor) [reply proposals/0289-result-builders.md:306]: Hmm. Sure, we can do that.

**Review by compnerd — APPROVED** [review]
Thanks for updating the proposal Doug!

**Pull request merged**